### DISCUSSION TITLE
Welcome to the py4pd Discussions!

### DISCUSSION THREAD

**Discussion by charlesneimog** [charlesneimog/py4pd#38] [General]
Welcome to the discussions for `py4pd`, a PureData object that integrates the power of [Python](https://www.python.org/) into the real-time environment of [PureData](https://puredata.info/). This space is dedicated to fostering collaboration, exchanging ideas, and providing support for all things related to `py4pd`.

Here, we can:

1. **Introduce Yourself**: Share your background in PureData, Python, or related fields.
2. **Ask Questions**: Seek guidance on `py4pd`'s integration with PureData.
3. **Share Ideas**: Discuss innovative techniques and approaches.
4. **Collaborate**: Showcase projects and collaborate with fellow enthusiasts.

Let's create an inclusive and supportive environment as we unlock the potential of `py4pd` in PureData.

Join the conversation and let's dive into the possibilities together!

Charles K. Neimog

**Comment by martindupras** [comment]
Hi! I'm Martin, very experienced with PD, quite new to Python. 

I'm very interested in trying out py4pd, which would be really useful to me!

I see that the docs say we can only use version 3.11 and not any other version. I get that; however I have version 3.12 installed (which as I understand it is a bug fix release.) Do I have to uninstall 3.12 and install 3.11 or is there a workaround? 

This is what I get in the PD console after installing from deken:

`[deken] Successfully installed 'py4pd'!
/Users/martindupras/Documents/Pd/externals/py4pd/py4pd.d_arm64:dlopen(/Users/martindupras/Documents/Pd/externals/py4pd/py4pd.d_arm64, 0x000A): Library not loaded: /Library/Frameworks/Python.framework/Versions/3.11/Python
  Referenced from: <A2692B49-EBBB-3683-BF34-C71FE77B888B> /Users/martindupras/Documents/Pd/externals/py4pd/py4pd.d_arm64
  Reason: tried: '/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/System/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file, not in dyld cache)
/Users/martindupras/Documents/Pd/externals/py4pd/py4pd.d_arm64:dlopen(/Users/martindupras/Documents/Pd/externals/py4pd/py4pd.d_arm64, 0x000A): Library not loaded: /Library/Frameworks/Python.framework/Versions/3.11/Python
  Referenced from: <A2692B49-EBBB-3683-BF34-C71FE77B888B> /Users/martindupras/Documents/Pd/externals/py4pd/py4pd.d_arm64
  Reason: tried: '/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/System/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file, not in dyld cache)
py4pd
... couldn't create`

I'm on MacOS Sonoma (14.2.1) with an M3.

Thanks!

**Comment by charlesneimog** [comment]
Hi Martin,

For Python, the third number is the bug release `3.11.7`. Python 3.11 and 3.12 are different.

You have two options: 
1. Install `python3.11`, which I recommend, mainly because there are some libraries that do not work in `python3.12` yet, `librosa`, and `loristrck`, among others.

2. Replace the `py4pd` package folder with `py4pd_0.8.5-beta_python3.12.zip` available [here](https://github.com/charlesneimog/py4pd/releases/tag/0.8.5-beta) 

The error you are seeing is because `py4pd` is trying to find `python3.11` and cannot found it.

Thanks for use `py4pd`, if you find bugs, please tell me!!!

Cheers

**Reply by martindupras** [reply]
Awesome. Thanks! I'll try the beta first!

Many thanks!

**Reply by martindupras** [reply]
I've ended up reinstalling python 3.11 and py4pd seems to work fine.
